src/Transformer/CustomTransformer/CustomTransformersRegistry.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\Transformer\CustomTransformer;
+
+use AutoMapper\MapperMetadataInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+final class CustomTransformersRegistry
+{
+    private array $customTransformers = [];
+
+    private array|null $prioritizedCustomTransformers = null;
+
+    public function addCustomTransformer(CustomTransformerInterface $customTransformer): void
+    {
+        if (!isset($this->customTransformers[$customTransformer::class])) {
+            $this->customTransformers[$customTransformer::class] = $customTransformer;
+        }
+    }
+
+    /**
+     * @param Type[] $sourceTypes
+     * @param Type[] $targetTypes
+     *
+     * @return null|class-string<CustomTransformerInterface>
+     */
+    public function getCustomTransformerClass(MapperMetadataInterface $mapperMetadata, array $sourceTypes, array $targetTypes, string $propertyName): string|null
+    {
+        // todo: what to do with union types?
```

Comment:
I don't understand exactly the issue here, could you tell me more about it ?

Reply:
I want the interface to stay simple. ie: I don't want it to be `public function supports(array<Type> $sources, array<Type> $targets): bool;` and to need to loop oever types

so, what to do if we map a union type to another union type, and several transformers supports the transformation?